fog/view/server/src/fog_view_router_service.rs

```diff
@@ -78,3 +86,63 @@ where
         });
     }
 }
+
+impl<E> FogViewApi for FogViewRouterService<E>
+where
+    E: ViewEnclaveProxy,
+{
+    fn auth(
+        &mut self,
+        ctx: RpcContext,
+        request: attest::AuthMessage,
+        sink: UnarySink<attest::AuthMessage>,
+    ) {
+        let _timer = SVC_COUNTERS.req(&ctx);
+        mc_common::logger::scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
+            if let Err(err) = check_request_chain_id(&self.chain_id, &ctx) {
+                return send_result(ctx, sink, Err(err), logger);
+            }
+
+            if let Err(err) = self.authenticator.authenticate_rpc(&ctx) {
+                return send_result(ctx, sink, err.into(), logger);
+            }
+            let result = router_request_handler::handle_auth_request(
+                self.enclave.clone(),
+                request,
+                self.logger.clone(),
+            )
+            .map(|mut response| response.take_auth());
+
+            send_result(ctx, sink, result, logger);
+        })
+    }
+
+    fn query(
+        &mut self,
+        ctx: RpcContext,
+        request: attest::Message,
+        sink: UnarySink<attest::Message>,
+    ) {
+        mc_common::logger::scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
```

Comment:
:wrench: Should this have a counters call like the `FogViewService` implementation?
```suggestion
    ) {
        let _timer = SVC_COUNTERS.req(&ctx);
        mc_common::logger::scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
```

Reply:
Yes it should have counters, but I think we should do this in a follow up PR that makes sure we're doing it correctly (might require some backend configuration etc)